internal/pkg/iptablesutils/iptables.go

```diff
@@ -32,6 +36,60 @@ const (
 	ModeLegacy = "legacy"
 )
 
+// ExtractIPTablesBinaries extracts the iptables binaries from the k0s binary and makes the symlinks
+// to the backend detected by DetectHostIPTablesMode.
+// ExtractIPTablesBinaries only works on linux, if called in another OS it will return an error.
+func ExtractIPTablesBinaries(k0sBinDir string, iptablesMode string) (error, string) {
+	if runtime.GOOS != "linux" {
```

Comment:
The first error will be staging the binaries and it will fail with: `return fmt.Errorf("unable to determine current executable: %w", err)` which isn't very intuitive to an end user,  but we should be able to spot that so I'll remove that.